projects/eslint-plugin/rules/general/lib/rules/no-conditional-object-keys.js

```diff
@@ -0,0 +1,57 @@
+/**
+ * SPDX-FileCopyrightText: © 2017 Liferay, Inc. <https://liferay.com>
+ * SPDX-License-Identifier: MIT
+ */
+
+const OBJECT = 'Object';
+const KEYS = 'keys';
+const EXCLAMATION = '!';
+
+const message = "don't use truthy values in conditionals";
+
+module.exports = {
+	create(context) {
+		const throwError = (node, objectName, propertyName) => {
+			if (objectName === OBJECT && propertyName === KEYS) {
+				context.report({
+					message,
+					node,
+				});
+			}
+		};
```

Comment:
`throwError` is a little strange for the name since it only reports an error if a condition is met. I would call this something like `checkExpression`.